ThirdPartyAdapters/unity/unity/src/main/java/com/google/ads/mediation/unity/UnityAdapter.java

```diff
@@ -190,60 +242,95 @@ public void requestInterstitialAd(Context context,
             }
             return;
         }
-        Activity activity = (Activity) context;
+        final Activity activity = (Activity) context;
         mActivityWeakReference = new WeakReference<>(activity);
 
-        UnitySingleton.getInstance().initializeUnityAds(activity, gameId);
-        UnitySingleton.getInstance().loadAd(mUnityAdapterDelegate);
-
+        UnityAds.addListener(mUnityAdapterDelegate);
+        UnitySingleton.getInstance().initializeUnityAds(activity, gameId,
+                new UnitySingleton.Listener() {
+                    @Override
+                    public void onInitializeSuccess() {
+                        MetaData metadata = new MetaData(activity);
+                        metadata.setCategory("mediation_adapter");
+                        metadata.set(uuid, "load-interstitial");
+                        metadata.set(uuid, mPlacementId);
+                        metadata.commit();
+
+                        UnitySingleton.getInstance().mPlacementsInUse.add(mPlacementId);
```

Comment:
should we set it before calling `initializeUnityAds`? also if it is in we should report no fill immideatly to be aligned with previous implementation.